libr/debug/debug.c

```diff
@@ -668,6 +668,26 @@ R_API const char *r_debug_reason_to_string(int type) {
 	case R_DEBUG_REASON_FPU: return "fpu";
 	case R_DEBUG_REASON_STEP: return "step";
 	case R_DEBUG_REASON_USERSUSP: return "suspended-by-user";
+	case R_DEBUG_REASON_SIGALRM: return "alarm-signal";
+	case R_DEBUG_REASON_SIGBUS: return "bus-error-signal";
+	case R_DEBUG_REASON_SIGCHLD: return "child-terminated-signal";
+	case R_DEBUG_REASON_SIGCONT: return "continue-signal";
+	case R_DEBUG_REASON_SIGFPE: return "arithemtic-error-signal";
+#ifdef __sparc
+	case R_DEBUG_REASON_SIGEMT: return "hardware-error-signal";
+#endif
+	case R_DEBUG_REASON_SIGHUP: return "terminal-hangup-signal";
+	case R_DEBUG_REASON_SIGILL: return "incorrectly-formed-machine-language-instruction-signal";
+	case R_DEBUG_REASON_SIGINFO: return "get-foreground-process-group-signal";
+	case R_DEBUG_REASON_SIGIO: return "I/O possible";
```

Comment:
```suggestion
```

sigIO is delivered when a fd permits IO, which is something really nasty to do because the way signals and threads work (and its usually ignored). so better delete it as a debug reason. having it as a signal is better